code/LeftAndMain.php

```diff
@@ -330,141 +307,19 @@ public function getClientConfig(): array
     {
         // Add WYSIWYG link form schema before extensions are applied
         $this->beforeExtending('updateClientConfig', function (array &$clientConfig): void {
+            $modalController = ModalController::singleton();
             $clientConfig['form'] = [
                 'EditorExternalLink' => [
-                    'schemaUrl' => $this->Link('methodSchema/Modals/EditorExternalLink'),
+                    'schemaUrl' => $modalController->Link('schema/EditorExternalLink'),
                 ],
                 'EditorEmailLink' => [
-                    'schemaUrl' => $this->Link('methodSchema/Modals/EditorEmailLink'),
+                    'schemaUrl' => $modalController->Link('schema/EditorEmailLink'),
                 ],
             ];
         });
         return parent::getClientConfig();
     }
 
-    /**
-     * Get form schema helper
-     *
-     * @return FormSchema
-     */
-    public function getFormSchema()
-    {
-        return $this->schema;
-    }
-
-    /**
-     * Set form schema helper for this controller
-     *
-     * @param FormSchema $schema
-     * @return $this
-     */
-    public function setFormSchema(FormSchema $schema)
-    {
-        $this->schema = $schema;
-        return $this;
-    }
-
-    /**
-     * Gets a JSON schema representing the current edit form.
-     */
-    public function schema(HTTPRequest $request): HTTPResponse
-    {
-        $formName = $request->param('FormName');
-        $itemID = $request->param('ItemID');
-
-        if (!$formName) {
-            $this->jsonError(400, 'Missing request params');
-        }
-
-        $formMethod = "get{$formName}";
-        if (!$this->hasMethod($formMethod)) {
-            $this->jsonError(404, 'Form not found');
-        }
-
-        if (!$this->hasAction($formName)) {
-            $this->jsonError(401, 'Form not accessible');
-        }
-
-        if ($itemID) {
-            $form = $this->{$formMethod}($itemID);
-        } else {
-            $form = $this->{$formMethod}();
-        }
-        $schemaID = $request->getURL();
-        return $this->getSchemaResponse($schemaID, $form);
-    }
-
-    /**
-     * Get the form schema from a given method.
-     * The method must return a Form.
-     */
-    public function methodSchema(HTTPRequest $request): HTTPResponse
```

Comment:
This method was just a bizarre way to avoid directly linking to `ModalController`.
It's not used anywhere now that we're linking to `ModalController` directly.